server/core/storage.go

```diff
@@ -222,29 +222,42 @@ func (s *Storage) LoadConfig(cfg interface{}) (bool, error) {
 	return true, nil
 }
 
-// SaveRules stores rule cfg to the rulesPath.
-func (s *Storage) SaveRules(rules interface{}) error {
+// SaveRule stores a rule cfg to the rulesPath.
+func (s *Storage) SaveRule(ruleKey string, rules interface{}) error {
 	value, err := json.Marshal(rules)
 	if err != nil {
 		return errors.WithStack(err)
 	}
-	return s.Save(rulesPath, string(value))
+	return s.Save(path.Join(rulesPath, ruleKey), string(value))
+}
+
+// DeleteRule removes a rule from storage.
+func (s *Storage) DeleteRule(ruleKey string) error {
+	return s.Base.Remove(path.Join(rulesPath, ruleKey))
 }
 
 // LoadRules loads placement rules from storage.
-func (s *Storage) LoadRules(rules interface{}) (bool, error) {
-	value, err := s.Load(rulesPath)
-	if err != nil {
-		return false, err
-	}
-	if value == "" {
-		return false, nil
-	}
-	err = json.Unmarshal([]byte(value), rules)
-	if err != nil {
-		return false, errors.WithStack(err)
+func (s *Storage) LoadRules(f func(v string) error) (bool, error) {
+	nextKey := path.Join(rulesPath, "\x00")
+	endKey := rulesPath + "0" // upper bound of all rules.
```

Comment:
I think we need some comment about `\x00` and `"0"`

Reply:
good idea